src/Components/WebView/WebView/src/WebViewManager.cs

```diff
@@ -204,25 +206,29 @@ record RootComponent
         /// Disposes the current <see cref="WebViewManager"/> instance.
         /// </summary>
         /// <param name="disposing"><c>true</c> when dispose was called explicitly; <c>false</c> when it is called as part of the finalizer.</param>
-        protected virtual void Dispose(bool disposing)
+        protected virtual async ValueTask DisposeAsync(bool disposing)
         {
             if (!_disposed)
             {
                 if (disposing)
                 {
-                    _currentPageContext?.Dispose();
+                    if (_currentPageContext != null)
+                    {
+                        await _currentPageContext.DisposeAsync();
+                    }
                 }
 
                 _disposed = true;
```

Comment:
If it were me I'd set this flag first and then return `_currentPageContext?.DisposeAsync() ?? default` but I probably try to avoid async state machines too much 😅

Reply:
This is the general pattern in [the docs](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-disposeasync), though this exact case of sync-calling-async isn't present.